pkg/pot/lexer.go

```diff
@@ -0,0 +1,115 @@
+package pot
+
+import (
+	"bufio"
+	"io"
+	"strconv"
+	"strings"
+)
+
+type TokenType int
+
+const (
+	HeaderLanguage TokenType = iota
+	HeaderTranslator
+	HeaderPluralForms
+	MsgCtxt
+	MsgId
+	MsgStr
+	PluralMsgId
+	PluralMsgStr
+	TranslatorComment
+	ExtractedComment
+	Reference
+	Flag
+	MsgctxtPreviousContext
+	MsgidPluralPrevUntStrPlural
+	MsgidPrevUntdStr
+)
+
+type Token struct {
+	Value string
+	Type  TokenType
+	Index int // plural index for msgstr with plural forms
+}
+
+func Lex(r *bufio.Reader) []Token {
+	var tokens []Token
+
+	for {
+		line, err := r.ReadString('\n')
+		if err == io.EOF {
+			break
+		} else if err != nil {
+			panic(err)
```

Comment:
I think we should return the error and not panic after all. 
It would be silly to terminate service if there is an error in the reading `.po` file.

Reply:
It's not gonna terminate the service, the server recovers from the panics and sends a trace to collectors. But you are right - return error.